### PR TITLE
DOCS-#6819: Update Modin on cluster documentation

### LINKED ISSUE #6819: Update Modin on cluster documentation

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #6819 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: docs/getting_started/using_modin/using_modin_cluster.rst [hunk 1]
```diff
@@ -31,8 +32,11 @@ To start up the Ray cluster, run the following command in your terminal:
 
 This configuration script starts 1 head node (m5.24xlarge) and 7 workers (m5.24xlarge),
 768 total CPUs. For more information on how to launch a Ray cluster across different
-cloud providers or on-premise, you can also refer to the Ray documentation `here <https://docs.ray.io/en/latest/cluster/cloud.html>`_.
+cloud providers or on-premise, you can also refer to the `Ray's cluster docs`_.
 
+.. note::
+   By default, Modin uses 60% of the system memory. It is recommended to use the same
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin_cluster.rst:38]:
```suggestion
   By default, Modin on Ray uses 60% of the system memory. It is recommended to use the same
```
Why 60%?

File: docs/getting_started/using_modin/using_modin_cluster.rst [hunk 2]
```diff
@@ -56,13 +60,20 @@ Modin:
 
 Congratualions! You have successfully connected to the Ray cluster.
 
+.. note::
+   Be careful when using the Ray client to connect to a remote cluster, most likely
+   this connection mode will not work. Known bugs:
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin_cluster.rst:65]:
```suggestion
   Be careful when using the Ray client to connect to a remote cluster.
   This connection mode may not work. Known bugs:
```

File: docs/getting_started/using_modin/using_modin_cluster.rst [hunk 3]
```diff
@@ -108,4 +123,10 @@ you can customize your Ray environment for use in Modin!
...
 .. _`pandas`: https://pandas.pydata.org/pandas-docs/stable/
 .. _`open an issue`: https://github.com/modin-project/modin/issues
-.. _`Ray's documentation`: https://ray.readthedocs.io/en/latest/api.html
+.. _`Ray's API docs`: https://ray.readthedocs.io/en/latest/api.html
+.. _`Part 1`: https://github.com/modin-project/modin/tree/master/examples/tutorial/jupyter/execution/pandas_on_ray/cluster/exercise_5.ipynb
+.. _`Part 2`: https://github.com/modin-project/modin/tree/master/examples/tutorial/jupyter/execution/pandas_on_ray/cluster/exercise_6.ipynb
+.. _`Ray's autoscaler options`: https://docs.ray.io/en/latest/cluster/vms/references/ray-cluster-configuration.html#cluster-config
+.. _`Ray's cluster docs`: https://docs.ray.io/en/latest/cluster/getting-started.html
+.. _`NYC Taxi dataset`: https://modin-datasets.s3.amazonaws.com/testing/yellow_tripdata_2015-01.csv
+.. _`Modin's cluster setup config`: https://github.com/modin-project/modin/blob/master/examples/tutorial/jupyter/execution/pandas_on_ray/cluster/modin-cluster.yaml
```
Comment (YarShev) [inline docs/getting_started/using_modin/using_modin_cluster.rst:132]: It seems this file is stale and we should update it as part of a separate issue. Actually, we would need update cluster modes for all distr backends - Ray, Dask, MPI.

File: modin/core/execution/ray/common/utils.py [hunk 4]
```diff
@@ -50,6 +50,8 @@
 _RAY_IGNORE_UNHANDLED_ERRORS_VAR = "RAY_IGNORE_UNHANDLED_ERRORS"
 
 ObjectIDType = ray.ObjectRef
+# TODO: Minimum version of Ray - 1.13
+# `if` branch can be deleted
```
Comment (YarShev) [inline modin/core/execution/ray/common/utils.py:54]: Good catch. We can also remove IsExperimental import in this file.
https://github.com/modin-project/modin/blob/324099d8737ead092e4acacfe03b1caa82e01986/modin/utils.py#L52C34-L52C48

File: docs/getting_started/using_modin/using_modin_cluster.rst (hunk 1 above)
Comment (anmyachev) [reply docs/getting_started/using_modin/using_modin_cluster.rst:38]: > Why 60%

https://github.com/modin-project/modin/blob/324099d8737ead092e4acacfe03b1caa82e01986/modin/core/execution/ray/common/utils.py#L44

File: modin/core/execution/ray/common/utils.py (hunk 4 above)
Comment (anmyachev) [reply modin/core/execution/ray/common/utils.py:54]: I wonder why `flake8` does not see that this import is not used. Hm

File: docs/getting_started/using_modin/using_modin_cluster.rst (hunk 3 above)
Comment (anmyachev) [reply docs/getting_started/using_modin/using_modin_cluster.rst:132]: > It seems this file is stale and we should update it as part of a separate issue.

Indeed. Maybe @Retribution98 can look at it? Here one will need to create a cluster in AWS.

> Actually, we would to need update cluster modes for all distr backends - Ray, Dask, MPI.

Don't get it.

**Review by YarShev — APPROVED** [review]

File: docs/getting_started/using_modin/using_modin_cluster.rst (hunk 3 above)
Comment (YarShev) [reply docs/getting_started/using_modin/using_modin_cluster.rst:132]: > > It seems this file is stale and we should update it as part of a separate issue.
> 
> Indeed. Maybe @Retribution98 can look at it? Here one will need to create a cluster in AWS.

We will schedule this offline.

> > Actually, we would to need update cluster modes for all distr backends - Ray, Dask, MPI.
> 
> Don't get it.

I mean docs/examples regarding Modin on cluster.

File: modin/core/execution/ray/common/utils.py (hunk 4 above)
Comment (anmyachev) [reply modin/core/execution/ray/common/utils.py:54]: @YarShev https://github.com/modin-project/modin/pull/6826

Comment (YarShev) [reply modin/core/execution/ray/common/utils.py:54]: thanks

**Pull request merged**